### DISCUSSION TITLE
How to obtain api-key on Google Cloud Run

### DISCUSSION THREAD

**Discussion by gazeto** [shlinkio/discussions#106] [Help wanted] [answered]
#### How Shlink is set up
* PHP Version: 8.2
* Shlink Version: docker image (latest)
* How do you serve Shlink: Docker image in Cloud Run
* Database engine used: MariaDB

#### Summary

Shlink and the web client are both running as a Docker image in Google Cloud Run. However, I have been trying for hours to **obtain the api-key**. GCR does not allow you to run docker `exec -it my_shlink shlink api-key:generate` in and didn't find any other way to do so.* A quick hint/help would be greatly appreciated.

*https://stackoverflow.com/questions/56342904/enter-a-docker-container-running-with-google-cloud-run

**Comment by acelaya** [comment, marked as answer]
Perhaps the `INITIAL_API_KEY` env var comes handy here.

The value you provide will be used to create an admin API key the first time a container is spin-up. After that you can even remove it if you want.

Make sure to set a secure value (long and random enough).

**Reply by acelaya** [reply]
More details here https://shlink.io/documentation/environment-variables/#server

**Comment by gazeto** [comment]
I can't believe I didn't find this earlier! (Maybe you could include a small reference/hint in the installation guide?)
**As expected, this works perfectly now.** Thank you so much!

_For reference for others_:
1. simply make up a random API key yourself
2. pass it to shlink with `INITIAL_API_KEY` with the initial run
3. Then run the web-client and enter your self-made API-key in the UI.

**Reply by acelaya** [reply]
> Maybe you could include a small reference/hint in the installation guide?

You are right. I need to check where does this better fit.

It should be probably mentioned at least here, I guess: https://shlink.io/documentation/install-docker-image/

**Reply by acelaya** [reply]
I guess this should cover it https://shlink.io/documentation/install-docker-image/#generate-first-api-key

**Reply by gazeto** [reply]
Yes, this is perfect!
